### PR TITLE
arrow: modernize more for conan v2 (take two)

### PR DESCRIPTION
**Pull request by mpimenov**
I suggest a revival of #15983 to use it as a starting point for further improvements. See discussion in the original PR.

### CODE REVIEW

**Comment by mpimenov** [comment]
I've rebased the commit to the latest master.

**Comment by mpimenov** [comment]
@Jean1995 Not yet but we'll get to it after this one is merged (or sooner).

**Review by RubenRBS — COMMENTED** [review]
Hi @mpimenov thanks for your contribution and your patience

> Can anyone take a look at this please?

It usually takes us a few days before we can properly review PRs with lots of changes :)

I have one minor question before merging, but the PR itself looks good for now!

File: recipes/arrow/all/conanfile.py [hunk 1]
```diff
@@ -539,81 +545,57 @@ def package(self):
         rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
         rmdir(self, os.path.join(self.package_folder, "share"))
 
-    def _lib_name(self, name):
-        if is_msvc(self) and not self.options.shared:
-            return "{}_static".format(name)
-        else:
-            return "{}".format(name)
+    def package_info(self):
+        # FIXME: fix CMake targets of components
```
[the remark this answers is not in the document]
Comment (mpimenov) [reply recipes/arrow/all/conanfile.py:549]: As the author explains in https://github.com/conan-io/conan-center-index/pull/15983,

> I'm pretty sure that CMake target names are wrong in `package_info()` (for `CMakeDeps` & `cmake_find_package[_multi]`), but I don't have the time to fix them.

Note that they are, well, moving targets: libX may become libarrow_X in another release. Here is an example of a similar fix: https://github.com/conan-io/conan-center-index/pull/17691

[the remark this answers is not in the document]
Comment (RubenRBS) [reply recipes/arrow/all/conanfile.py:549]: Thanks for the extra context :)

**Review by RubenRBS — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**